### DISCUSSION TITLE
Regarding pre-processing NLDAS2 forcing data for regional simulation

### DISCUSSION THREAD

**Discussion by SangamYa** [NCAR/hrldas#125] [Q&A]
I'm running into a snag while following the 'tutorial/Note2_2D_NLDAS2_domain.ipynb.' I'm trying to execute the model using the tutorial datasets available [here](https://drive.google.com/drive/folders/1ZghXJ2NApdFPdkOC-DKpSj_p7AT_yQ_0).

During the NLDAS data pre-processing, when I run the './create_forcing.exe namelist.input.NLDAS' command, I encounter the following error:

```bash
'Unknown geo_em map projection: map_proj = 32767'
```

Even though I'm using the provided `geo_em.d01_NLDAS0125.nc` file, it seems there's an issue with the map projection code (32767) not being recognized.

Any ideas on how I can resolve this issue? I'd appreciate any guidance or insights!

**Comment by cenlinhe** [comment]
@CharlesZheZhang Could you please look into this issue? Is it because of the geo_em file issue?

**Comment by CharlesZheZhang** [comment]
I have run the tutorial case on Cheyenne computer and don't have problem with reading the geo_em file map projection. 
When I print out the map_proj attribute from `geo_em.d01_NLDAS0125.nc` file, I have following result:
`map_proj =          6`
that is different from the above error message. 
I suspect this could be due to the netcdf library issue when reading the geo_em file, as read_geo_em_file(trim(geo_em_flnm), geo_em, ierr) is the first subroutine call that requires netcdf library.
https://github.com/NCAR/hrldas/blob/42627271cf3c19522e568311cc41fe3115ff12c3/hrldas/HRLDAS_forcing/create_forcing.F#L258C1-L258C57

**Reply by SangamYa** [reply]
Thank for the prompt response.
I am currently using 'netcdf-c-4.7.4' ,'netcdf-fortran-4.5.3' and have tried to compile the same in other pc's as well but the same error, I am getting: 'Unknown geo_em map projection:  map_proj = 32767'. Kindly suggest if there is any specific version of netcdf library needs to be installed?

**Reply by CharlesZheZhang** [reply]
Hello, we have three versions of netcdf depending on your fortran compiler version, these are the configuration used in the [configure](hrldas/configure) and [user_build_options](hrldas/arch):

(1) netcdf/4.8.1 compiled by intel/19.1.1 
(2) netcdf/4.7.4 compiled by pgi/20.4 
(3) netcdf/4.8.1 compiled by gnu/4.8.1 (gfortran)

Thanks!
Zhe

**Reply by SangamYa** [reply]
@CharlesZheZhang 
**My system configurations:**
ldd (GNU libc) 2.12
GCC version - 9.3.0
HRLDAS version - v5.0.0
For my system, What could be the compatible versions of both netcdf-c and netcdf-fortran? 

I have configured my netcdf-c 4.7.4 by using '~path/library/netcdf --enable-netcdf-4', will that be a issue? or is it only a version issue?
